### DISCUSSION TITLE
How can I share a .env file among all apps?

### DISCUSSION THREAD

**Discussion by NikhilCodes** [vercel/turborepo#593] [Help] [answered]
I have this `.env.local` file. Wondering how to share it among other apps.
Help!!

**Comment by albohlabs** [comment, marked as answer]
You can share it by the help of [dotenv-flow](https://github.com/kerimdzhanov/dotenv-flow) which you need to install first. Then you can move/merge all `.env.*` files to the top level folder and add the following to each `next.config.js` file.

```javascript
const path = require('path');

require('dotenv-flow').config({
  path: path.join(__dirname, '/../..'),
  node_env: process.env.NODE_ENV || 'development',
});

const env = {};
Object.keys(process.env).forEach((key) => {
  if (key.startsWith('NEXT_PUBLIC_')) {
    env[key] = process.env[key];
  }
});

module.exports = {
  env
}
```

**Reply by ataschz-g** [reply]
Do you have an example about this implementation?

**Reply by nilooy** [reply]
worked perfectly for me, Thanks a lot!

**Comment by bobbyflowstate** [comment]
This worked perfectly for me. Thank you @albohlabs !!
